### ISSUE TITLE
False negative for Lint/ImplicitStringConcatenation

### ISSUE THREAD

**Issue by Zopolis4** [rubocop/rubocop#13044]
## Expected behavior

To report this code as an offense and autocorrect it.

## Actual behavior

No offenses were reported.

## Steps to reproduce the problem

```ruby
# frozen_string_literal: true

foo = 'bar'
puts "#{foo}"'bar'
```

## RuboCop version

```
$ rubocop -V
1.65.0 (using Parser 3.3.4.0, rubocop-ast 1.31.3, running on ruby 3.1.2) [x86_64-linux-gnu]
```
